quakemigrate/io/data.py

```diff
@@ -578,141 +540,7 @@ def times(self, **kwargs):
         """
 
         # Utilise the .times() method of `obspy.Trace` objects
-        tr = Trace(header={"npts": self.signal.shape[-1],
+        tr = Trace(header={"npts": len(self.waveforms[0].data),
```

Comment:
It's only actually used in plotting and now that most of the data is kept in Trace objects, this is far less necessary. I suspect we can drop it and just fall back on the actual `Trace` routine, rather than 'reconstructing' it.

The full list of uses:
`quakemigrate/plot/event.py` - 2 instances - both should be possible to remove, at (what I expect to be) a negligible cost. Snippet: 
```
    # --- Waveforms ---
    mint, maxt = event.otime - 0.1, event.otime + max_tts*1.5
    for i, station in enumerate(event.data.stations):
        waveforms = event.data.filtered_waveforms.select(station=station)
        for c, comp in zip(WAVEFORM_COLOURS1, "ZNE"):
            data = waveforms.select(channel=f"*{comp}")
            if not bool(data):
                continue
            data = data[0].data
            times_utc = data[0].times(type="UTCDateTime")
            mint_i, maxt_i = [np.argmin(abs(times_utc - t)) for t in (mint, maxt)]
            times_plot = data[0].times(type="matplotlib")[mint_i:maxt_i]
            data[mint_i:]

            # Get station specific range for norm factor
            stat_maxt = event.otime + ttsf[i]*1.5
            norm = max(abs(data[mint_i:np.argmin(abs(times_utc - stat_maxt))]))

            y = data[mint_i:maxt_i] / norm + range_order[i]
            label = f"{comp} component" if i == 0 else None
            ax.plot(times_plot, y, c=c, lw=0.3, label=label, alpha=0.85)
```
There seems to be a random `data[mint_i:]` in there as well that I'm not sure is doing anything...

`quakemigrate/plot/phase_picks.py` - used once. I suspect this can now also be dropped, and simply use:
```
    times = signal[0].times(type="utcdatetime")
```
as this section is done on a station by station basis. Equally, can be moved inside a loop as per the solution above, though it's not perfect as the onset plotting relies on the timestamps being the same as for the data.

`quakemigrate/signal/pickers/gaussian.py` - this is the sticking point. It relies on the timings from the event, there is no other way to get it, currently...

...so, one possible solution is to store onsets as streams, rather than in the dict structure. A little bit of effort to ensure the header is set up, but the serving should be simple enough to adapt, and all the correct timing information is available as the input data to onset is in the form of a `Stream` of `Trace`s. Ultimately this seems at least somewhat sensible, though comes at some extra costs.

Reply:
So in that first one, `data` is not a `WaveformData` object; just a stream selected from `data.filtered_waveforms` - it's using the `times()` method directly on a trace from that stream, right?

I'm going to leave the `times` method, but just have it operate directly on `data.filtered_waveforms[0]`, without using the `sampling_rate` class variable, which is otherwise unused and unnecessary (and which I am therefore removing).

Reply:
Yes, exactly.

Ok, looks good as a solution for now.

Reply:
Actually, wait. `signal` is also available there (=`event.data.filtered_waveforms`). So can also just use `signal[0].times()` for that case. Unless I'm missing something. In which case I'll just bin the `times` method, because if it's not doing anything extra with the class variables then it's not necessary.

Reply:
Actually actually.. the more I read your summary of the instances of `event.data.times()` the more puzzled I am!

What version of `quakemigrate/plot/event.py` were you looking at when you found those lines? They don't exist in the current one..

Reply:
Oh, my bad I wasn't explicit - that snippet is my proposed solution.

And yes, in that case we should be able to drop the times method entirely.

Reply:
Ha, ok. With you. I think it's all sorted. Pushing shortly (turned into a fairly involved 'clean up')